### ISSUE TITLE
Feature: Support for tabs

### ISSUE THREAD

**Issue by jsenecal** [daniel-nagy/md-data-table#92]
Hi there,

I don't know if this is the right place to ask for this but anyways:
I'm looking to integrate md-data-table with tabs to filter the table data, and I'm wondering if there is a better way than creating 5+ tables in the same page.

Also, I would like the tabs to be in between `md-data-table-toolbar` and `md-data-table-container` but I dont know if it is possible to re-use `md-data-table-toolbar` for multiple tables...

Thanks for the help!

**Comment by daniel-nagy** [comment]
If you use `ng-repeat` for column headers and table rows you should be able to just change the value of the variables you are looping over to change the content in the table. So you would only need one table.

I think what I would do in your situation is create a template for a table and use `ng-include` or some other template method to include the table in each one of your tabs. Something like,

```html
<md-card>
  <md-tabs>
    <md-tab ng-controller="table-one-controller">
      <md-data-table-toolbar>
        <!-- toolbar content -->
      <md-data-table-toolbar>
      <md-data-table-container ng-include="my-table-template.html"></md-data-table-contianer>
    </md-tab>
    <md-tab ng-controller="table-two-controller">
      <md-data-table-toolbar>
        <!-- toolbar content -->
      <md-data-table-toolbar>
      <md-data-table-container ng-include="my-table-template.html"></md-data-table-contianer>
    </md-tab>
  </md-tabs>
</md-card>
```

Nothing is stoping you from putting the toolbar outside the `md-tabs` directive.

```html
<md-data-table-toolbar>
  <!-- toolbar content -->
<md-data-table-toolbar>
<md-tabs>
  <md-tab ng-controller="table-one-controller">
    <md-data-table-container ng-include="my-table-template.html"></md-data-table-contianer>
  </md-tab>
  <md-tab ng-controller="table-two-controller">
    <md-data-table-container ng-include="my-table-template.html"></md-data-table-contianer>
  </md-tab>
</md-tabs>
```

but changing its content when you switch tabs will be more work and you may need to use some css to mike it look nice.

**Comment by daniel-nagy** [comment]
@jsenecal where you able to achieve your desired layout?

**Comment by jsenecal** [comment]
I'm actually working on this. So far it does not seem as trivial as explained mainly because of the expected behavior of tables in MD - They should be animated... I'll try to work on a plunker or something...

**Comment by daniel-nagy** [comment]
If you can create a codepen I'll have a look at it. There is a ticket open for table transitions but I don't think I will have time to work on it in the near future.

**Comment by jsenecal** [comment]
So, after playing a bit with this, I realized that your examples are actually creating the tables in the DOM. What I was trying to achieve is to have the actual data source change with the tabs but keep the table. I also wanted to keep the swipe left/right animations.

The tabs im my implementation are actually to trigger some filtering on the server side so the table headers are staying the same...

**Comment by jsenecal** [comment]
also, I dont think you can add a controller to an <md-tab>:
>https://docs.angularjs.org/error/$compile/multidir?p0=ngController&p1=&p2=mdTab&p3=

**Comment by jsenecal** [comment]
Also, putting the table inside an `<md-table>` throws an error... see: http://codepen.io/anon/pen/gpVxPE?editors=101

**Comment by daniel-nagy** [comment]
> So, after playing a bit with this, I realized that your examples are actually creating the tables in the DOM. What I was trying to achieve is to have the actual data source change with the tabs but keep the table. I also wanted to keep the swipe left/right animations.

I don't believe this is possible using only one table. As you switch tabs the template will change allowing for the swipe gesture and triggering the animation. Which is why I suggested creating a template for a table then using `ng-include` to inject it into each tab. If you really wanted to use only one table you would have to use tabs as normal buttons and forgo the swipe and animation features.

> also, I dont think you can add a controller to an <md-tab>

You can put the controller on the `md-data-table-container` element or wrap the table in an additional element and put the controller on that.

> Also, putting the table inside an <md-tab> throws an error (TypeError: self.showProgress is not a function)

The `self.defer` function is being called before `self.showProgress` has a chance to get defined in the `mdTableProgress` directive. This isn't a big deal and will not break anything. To eliminate this error I will probably need to first check if the function is defined. `angular.isFunction(self.showProgress)`

**Comment by jsenecal** [comment]
Thanks @daniel-nagy, I<m going to try what has been done [here](http://stackoverflow.com/questions/27221222/separate-controller-per-tab-in-angular-material-w-ui-router) and see if its going to work for that use-case as well.
